### ISSUE TITLE
Error : Use of reserved word 'class

### ISSUE THREAD

**Issue by jaronwright** [notenoughneon/await-semaphore#2]
Hey guys,

I am getting the following error when running my tests - 

----------------------

PhantomJS 2.1.1 (Mac OS X 0.0.0) ERROR
  **SyntaxError: Use of reserved word 'class'**
  at webpack:///~/await-semaphore/index.js:2:0 <- src/test.ts:126348

20 06 2017 09:44:28.577:ERROR [PhantomJS 2.1.1 (Mac OS X 0.0.0)]: SyntaxError: Use of reserved word 'class'
at http://localhost:9877/base/src/test.ts?ae17abc58c999f7e7e07f0dae2b4e0f5f23983d9:126348

----------------------

It seems similar to the issue @ironbone had.

We are using angular 2, jasmine and karma.
angular: "2.3.1"
jasmine: "2.5.38"
karma: "1.2.0"
node: "6.0.42
typescript: "2.0.10"

We are compiling down to ES5 - And it looks like the await-semaphore is compiling down to ES6. 

For anyone else experiencing this issue we are looking into: 
https://www.npmjs.com/package/semaphore

----------------------
Working with @Trevael and @subatomicglue

**Comment by notenoughneon** [comment]
Yes, await-semaphore targets es6. Back when I wrote this, typescript didn't support async/await targeting es5. It looks like they added that feature in typescript 2.1.

What happens if you edit tsconfig.json to target es5?
